functions/index.ts

```diff
@@ -1,168 +1,105 @@
-import { Game as Game, IHttpResponse as IHttpResponse } from "./Modules";
-const allowedRequestParameters = JSON.parse(process.env.ALLOWED_REQUEST_PARAMETERS!);
+import { Game, GetGame, ModifyGame, DeleteGame, IHttpResponse, IDynamoObject, ListGames } from "./game"
 
 exports.handler = async (event: any, context: any, callback: any) => {
   switch (event.path) {
     case ("/getGame"):
-      await getGame(event.queryStringParameters["gameName"]);
+      let getGameData = DeserializeGameData({gameName: event.queryStringParameters["gameName"]});
+      callback(null, await GetGameHttpResponse(getGameData));
       break;
     case ("/listGames"):
-      await listGames();
+      callback(null, await ListGamesHttpResponse());
       break;
     case("/createGame"):
-      await createGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      let createGameData = DeserializeGameData(JSON.parse(event.body));
+      callback(null, await CreateGameHttpResponse(createGameData));
       break;
     case("/modifyGame"):
-      await modifyGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      let modifyGameData = DeserializeGameData(JSON.parse(event.body));
+      callback(null, await ModifyGameHttpResponse(modifyGameData));
       break;
     case ("/deleteGame"):
-      await deleteGame(JSON.parse(event.body).gameName);
+      let deleteGameData = DeserializeGameData(JSON.parse(event.body));
+      callback(null, await DeleteGameHttpResponse(deleteGameData));
       break;
     default:
-      HttpResponse({statusCode: 400, body: JSON.stringify("Unable to process request")});
+      callback(null, HttpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
       break;
-  }
-  
-  async function getGame(gameName: string) {
-    let params = {
-      TableName: table,
-      Key: {
-        gameName: gameName
-      }
-    };
-
-    try {
-      let response = await docClient.get(params).promise();
-      if (response.Item) {
-        let game = new Game(response.Item.gameName, response.Item.yearReleased, response.Item.genre, response.Item.console, response.Item.developer);
-        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
-      } else {
-        throw Error;
-     }
-    } catch (err) {
-      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
     }
+}  
+
+export async function GetGameHttpResponse(game: Game) {
+  let response = await GetGame(game);
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Item) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to get game."});
```

Comment:
Are you sure this case always represents a missing game? (I would explicitly test for an empty struct here)